flink-table-store-core/src/main/java/org/apache/flink/table/store/table/source/TableStreamingReader.java

```diff
@@ -88,13 +86,7 @@ public TableStreamingReader(
         if (predicate != null) {
             scan.withFilter(predicate);
         }
-        enumerator =
-                new ContinuousDataFileSnapshotEnumerator(
-                        table.location(),
-                        scan,
-                        new FullStartingScanner(),
-                        new DeltaFollowUpScanner(),
-                        null);
+        enumerator = ContinuousDataFileSnapshotEnumerator.create(table, scan, null);
```

Comment:
What if user set 'scan.mode' = 'from-timestamp' or 'latest' for this table? The base records wont' be read.